newa/cli.py

```diff
@@ -155,6 +165,34 @@ def cmd_event(ctx: CLIContext, errata_ids: list[str], compose_ids: list[str]) ->
     if not errata_ids and not compose_ids:
         raise Exception('Missing event IDs!')
 
+    def apply_mapping(string: str,
+                      mapping: Optional[list[str]] = None,
+                      regexp: bool = True) -> str:
+        # define default mapping
+        if not mapping:
+            mapping = [
+                r'\.GA$=',
+                r'\.Z\.(MAIN\+)?EUS$=',
+                r'RHEL-10\.0\.BETA=RHEL-10-Beta',
+                r'$=-Nightly',
+                ]
+        new_string = string
+        for m in mapping:
+            r = re.fullmatch(r'([^\s=]+)=([^=]*)', m)
+            if r:
+                pattern, value = r.groups()
+                if regexp and re.search(pattern, new_string):
+                    new_string = re.sub(pattern, value, new_string)
+                    ctx.logger.debug(
+                        f'Found match in {new_string} for mapping {m}, new value {new_string}')
+                if (not regexp) and new_string == pattern:
+                    new_string = value
+                    ctx.logger.debug(
+                        f'Found match in {new_string} for mapping {m}, new value {new_string}')
+            else:
+                raise Exception(f"Mapping {m} does not having expected format 'patten=value'")
```

Comment:
You can reverse the condition and save one indentation level:

```python
if not r:
  raise Exception(...)
```

Reply:
Updated